LanguageFeatures/Inline-classes/syntax_A02_t07.dart

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A rule for <inlineClassDeclaration> is added to the grammar,
+/// along with some rules for elements used in inline class declarations:
+///
+/// <inlineClassDeclaration> ::=
+///   'final'? 'inline' 'class' <typeIdentifier> <typeParameters>? <interfaces>?
+///   '{'
+///     (<metadata> <inlineMemberDeclaration>)*
+///   '}'
+///
+/// <inlineMemberDeclaration> ::= <classMemberDefinition>
+///
+/// @description Checks that an inline class can be declared as `final` and can
+/// be extended, implemented and a mixin on it can be declared in the same
+/// library where it is defined
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+final inline class FIC {
+  final int id ;
+  const FIC(this.id);
+}
+
+base class BC extends FIC {
```

Comment:
This would be a compile-time error: It is not possible to create a subtype of an inline class which isn't another inline class (so it can't be a class, a mixin, or a mixin class).

Reply:
Expect errors for all the other declarations for the same reason.